ironfish/src/rpc/routes/chain/getTransaction.ts

```diff
@@ -72,36 +86,44 @@ router.register<typeof GetTransactionRequestSchema, GetTransactionResponse>(
     }
     const hashBuffer = BlockHashSerdeInstance.deserialize(request.data.blockHash)
 
-    const block = await node.chain.getBlock(hashBuffer)
-    if (!block) {
+    const blockHeader = await node.chain.getHeader(hashBuffer)
+    if (!blockHeader) {
       throw new ValidationError(`No block found`)
     }
 
     // Empty response used for case that transaction not found
     const rawTransaction: GetTransactionResponse = {
       fee: '0',
       expiration: 0,
+      index: 0,
       notesCount: 0,
       spendsCount: 0,
       signature: '',
       notesEncrypted: [],
       mints: [],
       burns: [],
     }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
 
-    block.transactions.map((transaction) => {
+    transactions.map(({ transaction, initialNoteIndex }) => {
       if (transaction.hash().toString('hex') === request.data.transactionHash) {
         const fee = transaction.fee().toString()
         const expiration = transaction.expiration()
         const signature = transaction.transactionSignature()
         const notesEncrypted = []
 
+        let noteIndex = initialNoteIndex
         for (const note of transaction.notes) {
-          notesEncrypted.push(note.serialize().toString('hex'))
+          notesEncrypted.push({
+            noteIndex: noteIndex,
+            noteData: note.serialize().toString('hex'),
+          })
+          noteIndex++
         }
 
         rawTransaction.fee = fee
         rawTransaction.expiration = expiration
+        rawTransaction.index = initialNoteIndex
```

Comment:
This is not the transaction index. A transaction index is an index in the block. That's common word used by bitcoin and other cryptocurrencies.

Reply:
Rename as `initialNoteIndex` in the latest commit.